balancer-js/src/modules/pools/factory/types.ts

```diff
@@ -18,7 +19,9 @@ export interface ComposableStableCreatePoolParameters
 }
 
 export interface WeightedCreatePoolParameters extends CreatePoolParameters {
-  weights: BigNumberish[];
+  rateProviders: string[];
+  normalizedWeights: BigNumberish[];
+  salt?: BytesLike;
```

Comment:
Is there a good reason/benefit to leaving this optional?

Reply:
Yes, the salt is a bytes32 value that the user can pass to ensure the uniqueness of the address of the pool, let it optional makes the default value of the salt as HashZero(0x0...0), which is a value that does not affect the address hashing. The  mock weighted pool v4 was created using the HashZero as salt:
https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/deployments/tasks/20230320-weighted-pool-v4/index.ts#L55